internal/nix/nixprofile/profile.go

```diff
@@ -26,11 +28,52 @@ func ProfileListItems(
 	profileDir string,
 ) (map[string]*NixProfileListItem, error) {
 
-	lines, err := nix.ProfileList(writer, profileDir)
+	version, err := nix.Version()
 	if err != nil {
 		return nil, err
 	}
 
+	// if version is >= 2.17.0, we can use the json output
+	useJSON := vercheck.SemverCompare(version, "2.17.0") >= 0
```

Comment:
I am somewhat concerned about this failing in an unexpected way. The underlying function call will return:
```
// An invalid semantic version string is considered less than a valid one.
// All invalid semantic version strings compare equal to each other.
```

According to https://lazamar.co.uk/nix-versions/?channel=nixpkgs-unstable&package=nix, there may be some prerelease versions of nix that have non-standard semver versions.

However, these prerelease versions also appear to be rather old, and not present in newer releases of nix. So, maybe this is okay?

Reply:
I think this is OK. non semver versions will be considered older which defaults to the non --json.

I think as a follow up after this PR we could:

(only use the `--json` flag if we're sure version is higher (e.g. both are semver)) and then try to parse things both ways (try one and if it fails try the other)